### PR TITLE
[rivers] add new port

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by FrankXie05** [comment]
>A formatter is provided for rivers under the header rivers/format.hpp. It presumes that <fmt/format.hpp> can be found as an include. 

Maybe this need `fmt`.
https://github.com/brevzin/rivers#formatting

**Comment by jiayuehua** [comment]
> > A formatter is provided for rivers under the header rivers/format.hpp. It presumes that <fmt/format.hpp> can be found as an include.
> 
> Maybe this need `fmt`. [brevzin/rivers#formatting](https://github.com/brevzin/rivers#formatting)

I will add `fmt` as a feature.

**Comment by FrankXie05** [comment]
This should be a default feature. :)

File: ports/rivers/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
+  "default-features": [
+    "fmt"
+  ],
```
Comment (dg0yt) [inline ports/rivers/vcpkg.json:19]: No, this should not be a default feature.

It is turned off by default upstream:
https://github.com/brevzin/rivers/blob/cfbd4c3e0ca9fcde03075327d6dd628e57589342/CMakeLists.txt#L14

IMO it doesn't need to be offered at all. This port is header-only. If you users want to use fmt with rivers, they can add it to their own projects.

Comment (jiayuehua) [reply ports/rivers/vcpkg.json:19]: @FrankXie05 revert back and make fmt not default feature? just as @dg0yt suggested?

Comment (FrankXie05) [reply ports/rivers/vcpkg.json:19]: I'm not an expert on `rivers`, nor have I ever used `rivers`. However, if this is the default behavior of upstream, it should be turned off.

**Review by FrankXie05 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**